hoot-core/src/main/cpp/hoot/core/util/GeometryPainter.cpp

```diff
@@ -273,27 +272,18 @@ void GeometryPainter::drawPolygon(QPainter& pt, const OGRPolygon* polygon, const
     lpt->setMatrix(pt.matrix());
     lpt->setPen(pen);
     lpt->setBrush(brush);
-  }
-  else
-  {
-    lpt = &pt;
-  }
 
-  const OGRLinearRing* ring = polygon->getExteriorRing();
-  QPolygonF qp;
-  _convertRingToQPolygon(ring, qp, m);
+    const OGRLinearRing* ring = polygon->getExteriorRing();
+    QPolygonF qp;
+    _convertRingToQPolygon(ring, qp, m);
 
-  lpt->setPen(Qt::NoPen);
-  lpt->setBrush(brush);
-  lpt->drawPolygon(qp, Qt::WindingFill);
-
-  lpt->setPen(pen);
-  lpt->setBrush(Qt::NoBrush);
-  lpt->drawPolygon(qp);
```

Comment:
This section of code is now duplicated into the two different blocks in the `if/else` to make the whole thing easier to understand.  If there are interior rings, then the `QPainter` and `QImage` objects are created, used, and deleted.  Sometimes a diff makes it look more complicated than it is.